spec/unit/provider/package/apt_spec.rb

```diff
@@ -86,6 +86,34 @@
     provider.run_preseed
   end
 
+  context "When listing all instances" do
```

Comment:
I think this test is going a bit too deep because it is also testing things that happen in `dpkg.rb`.
I guess it would be easier just to stub the `instances` method on `Puppet::Type::Package::ProviderDpkg` to return the list that we need since we are non interested in this context on what `Puppet::Type::Package::ProviderDpkg.instances` is doing internally.

Something like:
```ruby

  describe ".instances" do
    before do
      allow(Puppet::Type::Package::ProviderDpkg).to receive(:instances).and_return([resource])
    end

    context "when package is manual marked" do
      before do
        allow(described_class).to receive(:aptmark).and_return("#{resource.name}\n")
      end

      it 'sets mark to manual' do
        expect(resource).to receive(:mark=).with(:manual)
        described_class.instances
      end
    end

    context 'when package is not manual marked ' do
      before do
        allow(described_class).to receive(:aptmark).and_return("")
      end

      it 'does not set mark to manual' do
        expect(resource).not_to receive(:mark=).with(:manual)
        described_class.instances
      end
    end
  end
```

Note that `resource` is created at the begging of the spec suite and is set to be used by the provider: https://github.com/puppetlabs/puppet/blob/master/spec/unit/provider/package/apt_spec.rb#L4-L17

Similar tests can be written for the other methods. Let me know if you need any help